### PR TITLE
feat!: integrate existing `catppuccinPalette` into repository

### PR DESCRIPTION
**Pull request by atticus-sullivan**
As discussed in #1 this is my upload of my `.dtx` file + necessary adjustments + add `build.lua` for easy building and packaging.

Still a Todo
- [x] updating the README.md
- [x] Supply `\Cpt<Flavor>` commands to change the active style/flavor
- [x] Write a github action which automatically executed `l3build ctan` to obtain the `.zip` for uploading at ctan.
- [x] talk about author(s)
- [x] relicense

### CODE REVIEW

File: catppuccinpalette.dtx
Comment (sgoudham) [inline catppuccinpalette.dtx:18]: The license that we prefer across the organisation is MIT, would you be willing to re-license this under the organisation as specified in our [LICENSE](https://github.com/catppuccin/latex/blob/main/LICENSE)

Comment (atticus-sullivan) [reply catppuccinpalette.dtx:18]: Oh yes I've forgotten this in my todos above, licensing and the author specified in the docs are still to be discussed.

**Comment by atticus-sullivan** [comment]
Ok, this is progressing. I already added an example for the github-action (not quite satisfied with the doubly-zipped artifact but I don't know of a nice way of resolving this).

So how do we proceed with the authors noted in the docs. Which ones should I add to the list and how should the by noted?

Also the README.md is still not updated and the current state of the package is not quite well tested (everyone is welcome to do so).

**Comment by sgoudham** [comment]
Hi @atticus-sullivan, so sorry about the wait on this - I completely forgot about this for a while until I checked all my GitHub notifications.

> I already added an example for the github-action (not quite satisfied with the doubly-zipped artifact but I don't know of a nice way of resolving this).

Unfortunately this is just a GitHub Actions :tm: moment and there's not a way around it.

> So how do we proceed with the authors noted in the docs. Which ones should I add to the list and how should the by noted?

I asked the original maintainer (@walshyb) how they felt about attribution and they were okay about not being included in the author list if all the files included were not their work. Of course, we will keep them in the README and list them as a past maintainer in the README if they are no longer interested in contributing / collaborating with yourself. 

This is their full response from discord:

image

> Also the README.md is still not updated and the current state of the package is not quite well tested (everyone is welcome to do so).

If you could update the instructions on the README on how to import the package manually, I'm happy to test it out.

Again, apologies for the wait on this and I hope you're still interested in contributing to this repository, thank you!

**Review by sgoudham — COMMENTED** [review]
As specified in the comment above, the author list may need to be changed and the README needs updated, but the other changes look good!

**Comment by atticus-sullivan** [comment]
Alright. Almost there, any idea how the assets (the images in the `README.md`) were generated? (while the colors are still the same, maybe we should just regenerate them just to be on the safe side).

Also please check if the Author information is fine now.

And at the moment the installation instruction refers to the artifact generated by the github action. Maybe making a release which directly contains the assets (not doubly zipped or so) and referencing that instead in the installation instructions would be nice. But we'd do that only after merging this PR right?

**Comment by walshyb** [comment]
@atticus-sullivan If I remember correctly, I converted each example PDF into an image (I think I used png?). Then I used Catppuccin's [catwalk](https://github.com/catppuccin/catwalk) to combine them to generate that image

**Comment by atticus-sullivan** [comment]
Alright, thanks :+1: added a Makefile which automates (re)building the assets (based on the examples).

**Comment by atticus-sullivan** [comment]
Then I think this looks pretty finished from my side. Please be certain that the author information is correct / the way you want to.

And also the thing with creating a release (on github) would be still open. But I think that's more a thing for later.

File: build.lua
Comment (sgoudham) [inline build.lua:1]: Is there a reason why we need this file? Sorry, it's not immediately obvious to me!

**Review by sgoudham — CHANGES_REQUESTED** [review]
Thanks for finishing up this PR! The author attribution on the README is great, does an authors field need to exist in the LaTeX files too? 

 Just a few comments from me and I think it should be fine to merge.

Also, have we discussed of or know of the process on how to "transfer" the package on CTAN to allow multiple maintainers (i.e. Catppuccin staff and yourself) to upload the package to CTAN?

File: build.lua
Comment (atticus-sullivan) [reply build.lua:1]: That specifies how to build (i.e. what the files are and what the project is called) for the `l3build` command.

File: .github/workflows/package.yml
Comment (atticus-sullivan) [reply .github/workflows/package.yml:17]: What do you mean with that?

Comment (sgoudham) [reply .github/workflows/package.yml:17]: To try and make this less surprising, I was hoping we could pin the version of texlive-latex-extra that's being installed

ref: https://askubuntu.com/questions/92019/how-to-install-specific-ubuntu-deb-packages-with-exact-version

**Comment by sgoudham** [comment]
> I think I can just when uploading specify additional maintainers:

Would this give the additional maintainer(s) the power to update the package too? e.g. could I under the catppuccin account upload a new version to CTAN? (if you specified that name in this list)

File: .github/workflows/package.yml
Comment (atticus-sullivan) [reply .github/workflows/package.yml:17]: Ok I never did something like that. Won't we run into problems once that specific version of the package isn't in the repos anymore? (I'm not using ubuntu as my daily driver so I don't know for sure if that happens).

But the only two components we rely on are `pdflatex` and `l3build`. I think it's very unlikely that they will change the "api".

**Comment by atticus-sullivan** [comment]
tbh I'm not completely sure. But the only thing I found about this would indicate yes (the question would be how to properly reference you).

image
(https://ctan.org/file/help/ctan/CTAN-upload-addendum)

Seems like the e-mail address plays a key role.

File: .github/workflows/package.yml
Comment (sgoudham) [reply .github/workflows/package.yml:17]: > But the only two components we rely on are pdflatex and l3build. I think it's very unlikely that they will change the "api".

Sorry if this is a basic question but is it not possible to install these separately? My experience with installing latex packages like this is it can take a lot of time since it's installing so many addons/packages.

**Comment by sgoudham** [comment]
> tbh I'm not completely sure. But the only thing I found about this would indicate yes (the question would be how to properly reference you).

I've just created the [catppuccin account on CTAN](https://www.ctan.org/home/catppuccin), could you please add this account to the additional maintainers of the package?

image

File: .github/workflows/package.yml
Comment (atticus-sullivan) [reply .github/workflows/package.yml:17]: Oh yes that's true. Latex installation always takes quite some time. I'll check if the components are in a smaller package (not the `-extra` one).
When I created the workflow it was also kinda the thought, well if the workflow runs for half a minute longer or not :shrug: 

Actually the workflows already run in [my repo/fork](https://github.com/atticus-sullivan/catppuccinLatex/actions) so at the moment the whole workflow runs in ~1 minute (and it's true, the installation takes up most of the time).

I have a workflow for another repo where I created a custom docker image which has latex already installed, but loading/fetching that one also takes ~ 30 seconds (but I'm really no docker expert so one might be able to squeeze some performance there).

**Comment by atticus-sullivan** [comment]
Alright, once merged I'll do the upload. There is also this field `Administrative notes (to the CTAN maintainers)` where I'll ask if the way we're/I'm doing this is correct and if that way you as the catppuccin orga will be able to submit updates in the end.

File: .github/workflows/package.yml
Comment (sgoudham) [reply .github/workflows/package.yml:17]: > So what do you think? Is is important to get the action execute faster than it is right now?

Thanks for some investigation into it. It's certainly not something to block the merge on but figured I'd point it out anyways that installing a full tex live distribution in CI/CD is probably something we'd like to avoid at some point in the future!

Comment (atticus-sullivan) [reply .github/workflows/package.yml:17]: Ok to be fair, there are github actions you can use to compile latex stuff, but that again doesn't work with l3build (I think) and it's a pain to do customization.

**Review by sgoudham — APPROVED** [review]
Think this is a good time to merge, thanks for the continued engagement and willingness to merge into our repository! Really appreciated. 

I'm happy to merge once you've accepted the invitation to the Catppuccin GitHub organisation and when you're free, could you please make a PR like the one listed below to track yourself as a current maintainer of this port. 

https://github.com/catppuccin/catppuccin/pull/2493/files

Thanks again!

**Comment by atticus-sullivan** [comment]
You're welcome. I see you're already working on that release thing. Should I wait until you're finished with uploading to ctan?

**Comment by sgoudham** [comment]
I'm just about to head to bed haha. I'll create the workflow to automatically create GitHub releases based on commits to the main branch tomorrow.

I can see that the current version of the package is 1.0.2 on CTAN. Am I correct in saying that this PR is essentially equivalent to 1.0.2 on CTAN? If so, I can tag this commit (https://github.com/catppuccin/latex/commit/bcdc7a56af7c1b16576480f2c553f246c919b944) as 1.0.2 and this repository can continue with semver versioning as new features/fixes/breaking changes are merged @atticus-sullivan

**Comment by atticus-sullivan** [comment]
Alright, I'll keep the webpage with the information filled in open for now then. (don't wonder though I might be hard to reach until the weekend)

No on CTAN is still the version from my other repository (not the fork of this one). Should be that version [here](https://github.com/atticus-sullivan/catppuccinPalette/releases/tag/v1.0.2) Until now I just said `1.1.0` would be the version migrated to the catppuccin orga (noted so in the `dtx`). But we could change so (would need to do so before the upload though).

**Pull request merged**
